corehq/ex-submodules/couchforms/getters.py

```diff
@@ -69,13 +72,32 @@ def get_instance_and_attachment(request):
     return instance, attachments
 
 
-def _valid_file_extension(file):
+def _valid_instance_file_extension(file):
     if "." not in file.name:
         return False
     file_extension = file.name.rsplit(".", 1)[-1]
     return file_extension == 'xml'
 
 
+def _valid_attachment_file(file):
+    return _valid_attachment_file_extension(file) or _valid_attachment_file_mimetype(file)
+
+
+def _valid_attachment_file_extension(file):
+    if "." not in file.name:
+        return False
+    file_extension = file.name.rsplit(".", 1)[-1]
+    return file_extension in VALID_ATTACHMENT_FILE_EXTENSIONS
```

Comment:
This code is almost identical to `_valid_instance_file_extension`. I think they could be combined and just pass in the list of acceptable extensions.

Reply:
Yes absolutely.

https://github.com/dimagi/commcare-hq/pull/32998/commits/21c4e89d4e196875723b631fed95e65a644ee048